src/components/pages/Triggers/TriggerDetails/TriggerSettings/General/Name.tsx

```diff
@@ -0,0 +1,64 @@
+import {Form} from 'antd';
+
+import {Input} from '@custom-antd';
+
+import {ConfigurationCard, notificationCall} from '@molecules';
+
+import {Permissions, usePermission} from '@permissions/base';
+
+import {useUpdateTriggerByIdMutation} from '@services/triggers';
+
+import {useStore} from '@store';
+
+import {displayDefaultNotificationFlow} from '@utils/notification';
+
+const Name: React.FC = () => {
+  const {currentTrigger, setCurrentTrigger} = useStore(state => ({
+    currentTrigger: state.currentTrigger!,
+    setCurrentTrigger: state.setCurrentTrigger,
+  }));
+
+  const mayEdit = usePermission(Permissions.editEntity);
+
+  const [updateTrigger] = useUpdateTriggerByIdMutation();
+
+  const [form] = Form.useForm();
+
+  const name = currentTrigger?.name;
+
+  const onFinish = () => {
+    const values = form.getFieldsValue();
+
+    const body = {
+      ...currentTrigger,
+      name: values.name,
+    };
+
+    return updateTrigger(body)
+      .then(res => displayDefaultNotificationFlow(res))
+      .then(() => {
+        notificationCall('passed', 'Trigger was successfully updated.');
+        setCurrentTrigger(body);
+      });
+  };
+
+  return (
+    <Form form={form} name="general-settings-name" initialValues={{name}} layout="vertical" disabled={!mayEdit}>
+      <ConfigurationCard
+        title="Trigger name"
+        description="Define the name of your trigger"
+        onConfirm={onFinish}
+        onCancel={() => {
+          form.resetFields();
+        }}
+        enabled={mayEdit}
```

Comment:
The form is not editable, and there is no need for `mayEdit` logic because of that too.

```suggestion
        isButtonsDisabled
        enabled={false}
        isEditable={false}
```